docs/virtual_machine_template_guide.adoc

```diff
@@ -0,0 +1,197 @@
+
+[source,yaml]
+----
+kind: Template
+metadata:
+  name: RHEL-7
+  labels:
+    miq.github.io/kubvirt/is-vm-template: true
+----
+
+=== Parameters
+
+Parameters allow a value to be supplied by the user or generated when the template is instantiated. +
+Then, that value is substituted wherever the parameter is referenced. +
+The following parameters will be used for the OfflineVirtualMachine name (and as a part of the persistent volume claim names) +
+Default values will be provided for memory and cpu, which will be used if the user does not supply a different value.
+
+[source,yaml]
+----
+parameters:
+- description: Name for the new VM     
+  name: NAME   
+- name: MEMORY
+  description: Amount of memory in MB
+  value: 512Mi  
+- name: CPU
+  description: Amount of cores
+  value: 4    
+----
+
+Additional info on link:https://kubernetes.io/docs/concepts/configuration/manage-compute-resources-container/[compute resources]
+
+
+== Creating the Template object in the cluster
+
+If you have a YAML file that defines a VirtualMachine template, you can upload the template to an Openshift cluster using the CLI. 
+To upload a template to your current project’s template library, pass the YAML file with the following command: +
+
+----
+$ oc create -f <filename>
+----
+
+
+=== Creating VirtualMachineTemplate Objects using the Web Console
+TBD  (add screenshots)
+
+=== Generating the VirtualMachineTemplate Objects from CLI 
+The list of parameters that you can override are listed under  Parameters.
+You can list them with the CLI by using the following command and specifying the file to be used:
+----
+  $ oc process --parameters -f <filename>
+----
+
+Or create objects from a template by processing the template and piping the output to oc create:
+
+----
+  $ oc process --parameters -f <filename> | oc create -f -
+----
+
+
+== Modifying an Uploaded Template
+You can edit a template that has already been uploaded to your project by using the following command: +
+
+----
+$ oc edit template <template>
+----
+
+
+
+=== Full VirtualMachine Template example
+
+[source,yaml]
+----
+apiVersion: kubevirt.io/v1alpha1
+kind: Template 
+metadata:
+  name: RHEL-7  #This will define the template’s name
+  # name: Microsoft-Windows-2012r1
+  labels:
+    miq.github.io/kubvirt/is-vm-template: true  
+objects:
+- apiVersion: kubevirt.io/v1alpha1
+  kind: OfflineVirtualMachine
+  metadata:
+    name: rhel-7-${NAME} 
+  spec:
+    template:
+      metadata:
+        labels:
+          my: label
+      spec:
+        domain:
+          resources:
+            requests:
+              memory: ${MEMORY} 
+          cpu:
+            cores: ${CPU}
+          devices:
+            disks:
+              name: disk0
+              volumeName: mypvc 
+      volumes:
+       - name: mypvc
+         presistentVolumeClaim:
+           claimName: rhel-7-pvc-${NAME}
+- apiVersion: kubevirt.io/v1alpha1
+  kind: PersistentVolumeClaim
+  metadata:
+    name: rhel-7-pvc-${NAME}       
```

Comment:
Here we also need to include the annotation to request a clone @aglitke - got an example?

Reply:
The previous is a non blocking item for merging this - but we should then add a FIXME note